app_check/integration_test/src/integration_test.cc

```diff
@@ -821,4 +919,191 @@ TEST_F(FirebaseAppCheckTest, TestFunctionsFailure) {
                     firebase::functions::kErrorUnauthenticated);
 }
 
+TEST_F(FirebaseAppCheckTest, TestFirestoreSetGet) {
+  InitializeAppCheckWithDebug();
+  InitializeApp();
+  InitializeFirestore();
+
+  firebase::firestore::DocumentReference document = CreateFirestoreDoc();
+
+  WaitForCompletion(
+      document.Set(firebase::firestore::MapFieldValue{
+          {"str", firebase::firestore::FieldValue::String("foo")},
+          {"int", firebase::firestore::FieldValue::Integer(123)}}),
+      "document.Set");
+  firebase::Future<firebase::firestore::DocumentSnapshot> future =
+      document.Get(firebase::firestore::Source::kServer);
+  WaitForCompletion(future, "document.Get");
+  EXPECT_NE(future.result(), nullptr);
```

Comment:
nit: `EXPECT_NE` -> `ASSERT_NE` to avoid a segfault on the next line in the case that `future.result()` erroneously returned `nullptr`.

Reply:
Ah, good catch